feat: Enhance Product Card and Implement Dynamic Routes

irtassedat opened the pull request:
- Enhanced ProductCard component to display API data dynamically.
- Implemented dynamic routing for product details page using `/:category/:productId/:productNameSlug` pattern.
- Identified an issue with carousel displaying duplicate images. Plan to address in a future update.

Note: This commit partially addresses the enhancements outlined in #24. Further improvements will be made in subsequent updates.

Ref #24

irtassedat committed:
feat: Enhance Product Card and Implement Dynamic Routes

- Enhanced ProductCard component to display API data dynamically.
- Implemented dynamic routing for product details page using `/:category/:productId/:productNameSlug` pattern.
- Identified an issue with carousel displaying duplicate images. Plan to address in a future update.

Note: This commit partially addresses the enhancements outlined in #24. Further improvements will be made in subsequent updates.

Ref #24

irtassedat commented:
> Bu pull request, Ürün Detay Sayfasına bir dizi iyileştirme getirerek kullanıcı deneyimini ve kod tabanı kalitesini artırmaya odaklanmaktadır:

### Carousel Komponenti: 

> Tekrar eden görselleri destekleyecek şekilde geliştirildi, böylece tüm ürün sayfalarında sorunsuz geçişler ve tutarlı bir kullanıcı deneyimi sağlanıyor.


### Dinamik Yönlendirme: 

> Daha iyi gezinme ve SEO uygulamalarını kolaylaştırmak için (/:category/:productId/:productNameSlug) dinamik yönlendirme uygulandı.

### Kod Refaktörü:

> findDOMNode kullanımının kaldırılması uyarıları ele alındı,ürüne ait aynı resmi göstermeye çalışırken benzersiz{key} uyarıları dikkate alındı, gelecekteki güncellemeler için React.StrictMode uyumluluğuna yönelik bir temel oluşturuldu.

**Daha fazla iyileştirme veya öneri için geri bildiriminizi bekliyorum.**

Refs #24

gokhanozdemir commented on src/components/HomePage/Carousel.jsx:
bunu constructor ile neden yaptın? :D gpt mi? Bu çok eski bir react. Yazdığını anladıysan sorun yok ama.

irtassedat replied:
carouselimi değiştirmek için şimdi bilgisayar başına oturdum @gokhanozdemir

gokhanozdemir replied:
Saat geç hadi yat :D

gokhanozdemir reviewed (approved):
GPT'yi asistanın olarak gör. Bu kodu açıklayabiliyorsan sorun yok. Nasıl çalıştığını mülakatta anlatamazsan, sonra bir ara baştan yapmanı öneririm.

gokhanozdemir commented:
Genel olarak ok merge edebilirsin

irtassedat committed:
T12: Fixed carousel duplication issue and implemented dynamic product details routing.

- Resolved the problem where carousel displayed duplicate images.
- Added dynamic routing for product details page using the /:category/:productId/:productNameSlug pattern.
- This commit addresses the issues and enhancements mentioned in #24. Further improvements will be continued in upcoming updates.

Closes #24.

irtassedat committed:
Removed unused expressions and further refined dynamic routing and carousel.

- Cleaned up the code by removing unused expressions and variables.

irtassedat commented:
> GPT'yi asistanın olarak gör. Bu kodu açıklayabiliyorsan sorun yok. Nasıl çalıştığını mülakatta anlatamazsan, sonra bir ara baştan yapmanı öneririm.

@gokhanozdemir sizin önerdiğiniz videoya uygun şekilde düzenledim, korktum anlatamamaktan istediğim gibi de çalışmıyordu zaten.. 💯
